Sources/SymbolKit/SymbolGraph/Symbol/Symbol.swift

```diff
@@ -281,92 +283,55 @@ extension SymbolGraph.Symbol {
-                case .deinit         : return "deinit"
-                case .enum           : return "enum"
-                case .case           : return "enum.case"
-                case .func           : return "func"
-                case .operator       : return "func.op"
-                case .`init`         : return "init"
-                case .method         : return "method"
-                case .property       : return "property"
-                case .protocol       : return "protocol"
-                case .struct         : return "struct"
-                case .subscript      : return "subscript"
-                case .typeMethod     : return "type.method"
-                case .typeProperty   : return "type.property"
-                case .typeSubscript  : return "type.subscript"
-                case .typealias      : return "typealias"
-                case .var            : return "var"
-                case .module         : return "module"
-                case .unknown        : return "unknown"
-                }
-            }
-        }
-
-        // FIXME: Save "unknown" symbol kinds in a synchronized set to prevent loss of data (rdar://84276085)
+    public struct KindIdentifier: Equatable, Hashable, Codable {
+        public let identifier: String
 
-        /// Check the given identifier string against the list of known identifiers.
-        ///
-        /// - Parameter identifier: The identifier string to check.
-        /// - Returns: The matching `KindIdentifier` case, or `nil` if there was no match.
-        private static func lookupIdentifier(identifier: String) -> KindIdentifier? {
-            switch identifier {
-            case "associatedtype" : return .associatedtype
-            case "class"          : return .class
-            case "deinit"         : return .deinit
-            case "enum"           : return .enum
-            case "enum.case"      : return .case
-            case "func"           : return .func
-            case "func.op"        : return .operator
-            case "init"           : return .`init`
-            case "method"         : return .method
-            case "property"       : return .property
-            case "protocol"       : return .protocol
-            case "struct"         : return .struct
-            case "subscript"      : return .subscript
-            case "type.method"    : return .typeMethod
-            case "type.property"  : return .typeProperty
-            case "type.subscript" : return .typeSubscript
-            case "typealias"      : return .typealias
-            case "var"            : return .var
-            case "module"         : return .module
-            default               : return nil
-            }
+        private init(rawIdentifier: String) {
+            self.identifier = rawIdentifier
         }
 
+        public static let `associatedtype`  : Self = .init(rawIdentifier: "associatedtype")
```

Comment:
Should we just use KindIdentifier(rawIdentifier:)? 

Which fits the swift-docc repo style better?

```suggestion
        public static let `associatedtype` = KindIdentifier(rawIdentifier: "associatedtype")
```